gen/gen.go

```diff
@@ -94,32 +113,32 @@ func Struct(s string, dst io.Writer, gc Config) error {
 	return err
 }
 
-func generateFrom(schema avro.Schema, acc *data) string {
+func (gc Config) generateFrom(schema avro.Schema, acc *data) string {
 	switch t := schema.(type) {
 	case *avro.RecordSchema:
 		typeName := strcase.ToCamel(t.Name())
 		fields := make([]field, len(t.Fields()))
 		for i, f := range t.Fields() {
 			fSchema := f.Type()
 			fieldName := strcase.ToCamel(f.Name())
-			typ := resolveType(fSchema, f.Prop("logicalType"), acc)
+			typ := gc.resolveType(fSchema, f.Prop("logicalType"), acc)
```

Comment:
Does that work? `logicalType` is reserved and should never end up in props.

Reply:
Ah, I see the issue. The logical type cannot be on the field, it must be on the fields schema.

Reply:
I missed it in the last review, a logical type would be:
```
{"name":"myField","type":{"type":"long","logicalType": "timestamp-micros"}}
```
You can get a logical type generically by casting it to a `LogicalSchema` and calling `Logical`.

Reply:
The issue still eludes me. Do I need to change something here? I reinspected these with a debugger and the only place I can find the logical type is on the field props, but I might be missing something

Reply:
Ah! Does lead me to ask: shouldn't the `avro.Parse` fail?

Reply:
Your previous comment https://github.com/hamba/avro/pull/162#discussion_r895969602 didn't appear to me until just now, sorry for asking twice, it was clear on the first explanation :+1: